doctr/models/modules/layers/tensorflow.py

```diff
@@ -93,3 +100,76 @@ def call(self, x: tf.Tensor, **kwargs: Any) -> tf.Tensor:
         id_out = self.rbr_identity(x, **kwargs) if self.rbr_identity is not None and self.ver_bn is not None else 0
 
         return self.activation(main_outputs + vertical_outputs + horizontal_outputs + id_out)
+
+    # The following logic is used to reparmetrize the layer
+    # Adapted from: https://github.com/mindee/doctr/blob/main/doctr/models/modules/layers/pytorch.py
+    def _identity_to_conv(
+        self, identity: layers.BatchNormalization
+    ) -> Union[Tuple[tf.Tensor, tf.Tensor], Tuple[int, int]]:
+        if identity is None or not hasattr(identity, "moving_mean") or not hasattr(identity, "moving_variance"):
+            return 0, 0
+        if not hasattr(self, "id_tensor"):
+            input_dim = self.in_channels // self.groups
+            kernel_value = np.zeros((self.in_channels, input_dim, 1, 1), dtype=np.float32)
+            for i in range(self.in_channels):
+                kernel_value[i, i % input_dim, 0, 0] = 1
+            id_tensor = tf.constant(kernel_value, dtype=tf.float32)
+            self.id_tensor = self._pad_to_mxn_tensor(id_tensor)
+        kernel = self.id_tensor
+        std = tf.sqrt(identity.moving_variance + identity.epsilon)
+        t = tf.reshape(identity.gamma / std, (-1, 1, 1, 1))
+        return kernel * t, identity.beta - identity.moving_mean * identity.gamma / std
+
+    def _fuse_bn_tensor(self, conv: layers.Conv2D, bn: layers.BatchNormalization) -> Tuple[tf.Tensor, tf.Tensor]:
+        kernel = conv.kernel
+        kernel = self._pad_to_mxn_tensor(kernel)
+        std = tf.sqrt(bn.moving_variance + bn.epsilon)
+        t = tf.reshape(bn.gamma / std, (1, 1, 1, -1))
```

Comment:
Why is it `(1, 1, 1, -1)` instead of `(-1, 1, 1, 1)` ?

Reply:
the shapes in TF and PT are different ^^

TF: (kernel_size, kernel_size, dim, filter_dim)
PT: (filter_dim, dim, kernel_size, kernel_size)